testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -95,6 +95,12 @@ def test_verbose(u):
     assert a._verbose
 
 
+def test_verbose_run(u):
```

Comment:
Here it would be good to test multiple cases using pytest.mark.parametrize, i.e. the cases of:

init, run
True, True
False, True
True, False
False, False

Let us know if you need any help to work out how to use parametrize. See `test_AnalysisFromFunction` for an example of it.

Reply:
@pytest.mark.parametrize('verbose, verbose_r',[(True,True), (True,False), (False,True), (False,False)])
def test_verbose_run(u,verbose, verbose_r)
   a = FrameAnalysis(u.trajectory, verbose=verbose).run(verbose = verbose_r)
(should I include the four conditions )  
 if (a._verbose == True and a._pm.verbose == True): 
      assert a._verbose
      assert a._pm.verbose
   elif (a._verbose == True and a._pm.verbose == False): 
      assert a._verbose
      assert not a._pm.verbose
   elif (a._verbose == False and a._pm.verbose == True): 
      assert not a._verbose
      assert a._pm.verbose
   elif (a._verbose == False and a._pm.verbose == False): 
      assert not a._verbose
      assert not a._pm.verbose
or simply 
      assert a._verbose
      assert a._pm.verbose

Reply:
You don't need the if/else construct, just assert that the a._verbose and a._pm.verbose values match the expected inputs verbose and verbose_r.

Obviously, this is all dependent on https://github.com/MDAnalysis/mdanalysis/pull/2631#issuecomment-601640537